cli/internal/daemon/lifecycle.go

```diff
@@ -3,92 +3,46 @@ package daemon
 import (
 	"context"
 	"fmt"
-	"github.com/vercel/turbo/cli/internal/config"
+	"github.com/vercel/turbo/cli/internal/turbostate"
 
 	"github.com/pkg/errors"
-	"github.com/spf13/cobra"
 	"github.com/vercel/turbo/cli/internal/cmdutil"
 	"github.com/vercel/turbo/cli/internal/daemon/connector"
 	"github.com/vercel/turbo/cli/internal/turbodprotocol"
 )
 
-func addStartCmd(root *cobra.Command, helper *cmdutil.Helper) {
-	cmd := &cobra.Command{
-		Use:           "start",
-		Short:         "Ensures that the turbo daemon is running",
-		SilenceUsage:  true,
-		SilenceErrors: true,
-		RunE: func(cmd *cobra.Command, args []string) error {
-			flags := config.FlagSet{FlagSet: cmd.Flags()}
-			base, err := helper.GetCmdBase(flags)
-			if err != nil {
-				return err
-			}
-			l := &lifecycle{
-				base,
-			}
-			if err := l.ensureStarted(cmd.Context()); err != nil {
-				l.logError(err)
-				return err
-			}
-			return nil
-		},
+// RunLifecycle executes the lifecycle commands `start`, `stop`, `restart`.
+func RunLifecycle(ctx context.Context, helper *cmdutil.Helper, args *turbostate.ParsedArgsFromRust) error {
```

Comment:
Was there a reason for grouping the 3 lifecycle commands into one function?

Reply:
Mostly each function was doing a very similar task and there was going to be an if statement regardless. I'm open to splitting it up if you think it's cleaner.

Reply:
If there needs to be an if statement then keeping it as-is is fine. A lot of these comments are just me asking questions for changes where I want to make sure there isn't something more happening that I'm not picking up on.